identity-iota/src/credential/validator.rs

```diff
@@ -87,53 +115,129 @@ impl<'a, R: TangleResolve> CredentialValidator<'a, R> {
   pub async fn validate_credential<T>(
     &self,
     credential: Credential<T>,
-    options: VerifierOptions,
+    signature_verification_options: VerifierOptions,
+    allowed_deficiencies: CredentialDeficiencySet,
   ) -> Result<CredentialValidation<T>>
   where
     T: Serialize,
   {
+    // setup deficiency event handler
+    let mut encountered_deficiencies = CredentialDeficiencySet::empty();
```

Comment:
Do we actually need to return that when the caller has already stated which deficiencies are (not) allowed?

Reply:
I think we do since the caller can only specify the kind of deficiency, but not be more precise than that. Maybe it would be better to use some other collection for recording the encountered allowed `Deficiencies` that also keeps track of the specifics however.

Reply:
Then again, as you pointed out, the caller can look at the presentation/credential themselves to see, e.g. whether they accept an expiration one day ago.